contracts/finance/PaymentSplitterDeployer.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2021 the ethier authors (github.com/divergencetech/ethier)
+pragma solidity >=0.8.0 <0.9.0;
+
+import "./ProxiedPaymentSplitter.sol";
+import "@openzeppelin/contracts/proxy/Clones.sol";
+
+/**
+@notice This contract deploys a modified version of OpenZeppelin's
+PaymentSplitter that can be used with a minimal proxy contract, and exposes
+functions to deploy such proxies to random or deterministic addresses.
+@dev As only a single instance of this contract need exist, the addresses of
+deployments will be published on github.com/divergencetech/ethier.
+ */
+contract PaymentSplitterDeployer {
+    using Clones for address;
+
+    /**
+    @notice The primary instance of the modified PaymentSplitter, delegated to
+    by all clones.
+     */
+    address public immutable primary;
+
+    constructor() {
+        primary = address(new ProxiedPaymentSplitter());
+
+        // The primary instance also needs to be initialised even though it will
+        // likely never be used.
+        address[] memory payees = new address[](1);
+        payees[0] = msg.sender;
+        uint256[] memory shares = new uint256[](1);
+        shares[0] = 1;
+
+        _postDeploy(primary, payees, shares);
```

Comment:
That was my initial thought too, but I figured that if we didn't then someone else would and it's safer to just do it than think through all the possibly malicious social engineering that could occur. What do you think?

Reply:
Had a quick look on how other people do this (e.g. https://github.com/OpenZeppelin/workshops/blob/master/02-contracts-clone/contracts/1-ERC20/FactoryClone.sol). They also seem to just ignore it. IMO, there's no real vector of attack here, because it has no effect on the clones and devs using proxied payment splitters are responsible for using the correct clone address anyways. Also inititalizing it, adds deployment gas, which can be avoided.